app/javascript/gobierto_visualizations/webapp/containers/contracts/ContractsShow.vue

```diff
@@ -132,7 +227,26 @@ export default {
       this.contract_type = contract_type
       this.permalink = permalink
       this.assignee_routing_id = assignee_routing_id
+      this.contractor = contractor
+      this.contract_type = contract_type
+      this.start_date = start_date
+      this.end_date = end_date
+      this.award_date = award_date
+      this.batch_number = +batch_number
+      this.minor_contract = minor_contract
     }
+
+    if (this.hasBatch) this.groupBatchs()
+  },
+  methods: {
+    groupBatchs() {
+      this.filterContractsBatchs = this.contractsData.filter(({ title }) => title === this.title).sort((a, b) => a.batch_number - b.batch_number);
+    },
+    calculatePercentage(value) {
+      const filterByContractor = this.contractsData.filter((contract) => contract[value] === this[value])
+      const totalAmount = filterByContractor.map(({ final_amount_no_taxes }) => final_amount_no_taxes).reduce((prev, next) => prev + next);
```

Comment:
You might save one operation:
```suggestion
      const totalAmount = filterByContractor.reduce((acc, { final_amount_no_taxes }) => acc + final_amount_no_taxes);
```

Remember, `filter`, `map`, `reduce` run through the whole array, so its time complexity is **O(n)**. Less operations you do, faster it will run.

Reply:
Done!